libevm/VM.cpp

```diff
@@ -815,7 +848,68 @@ void VM::interpretCases()
 			m_sp -= 2;
 		CASE_END
 
+#ifdef EVM_JUMPS_AND_SUBS
+		CASE_BEGIN(JUMPTO)
+			ON_OP();
+			updateIOGas();
+			m_pc = decode_jump_dest(m_code, m_pc);
+		CASE_END
+
+		CASE_BEGIN(JUMPIF)
+			ON_OP();
+			updateIOGas();
+			if (*m_sp)
+			{
+				m_pc = decode_jump_dest(m_code, m_pc);
+			}
+			else
+				++m_pc;
+			--m_sp;
+		CASE_END
+
 		CASE_BEGIN(JUMPV)
+			ON_OP();
+			updateIOGas();			
+			m_pc = decode_jumpv_dest(m_code, m_pc, m_sp);
+		CASE_END
+
+		CASE_BEGIN(JUMPSUB)
+			ON_OP();
+			updateIOGas();
+			{
+				*++m_rp = m_pc++;
```

Comment:
To be honest these _pointer tricks_ annoy me a bit. Can't we make it more explicit using vectors, push_backs, etc?